lighthouse/src/main.rs

```diff
@@ -38,47 +52,187 @@ fn main() {
+
+    let file_name_opt = cli_matches.value_of(CONFIG_FILE_FLAG);
+
+    if let Some(file_name) = file_name_opt {
+        let file_config = clap_utils::parse_file_config(file_name)?;
+
+        let mut file_args = HashMap::new();
+        for (key, value) in file_config.iter() {
+            file_args.insert(&**key, &**value);
+        }
+
+        cli_matches = new_app(version.as_str(), long_version.as_str(), Some(&file_args))
+            .get_matches_from(args)?;
+    };
+    Ok(cli_matches)
+}
+
+/// Instantiate a new `App<'a>`, using the given map of file args as fallback config.
+///
+/// Note: `App<'a>` here is actually the `clap_utils::DefaultConfigApp<'a>` wrapper around `cli::App<'a>`.
+fn new_app<'a>(
+    version: &'a str,
+    long_version: &'a str,
+    file_args: Option<&'a HashMap<&'a str, &'a str>>,
+) -> App<'a> {
+    App::new("Lighthouse", file_args)
+        .version(version)
         .author("Sigma Prime <contact@sigmaprime.io>")
-        .setting(clap::AppSettings::ColoredHelp)
         .about(
             "Ethereum 2.0 client by Sigma Prime. Provides a full-featured beacon \
              node, a validator client and utilities for managing validator accounts.",
         )
         .long_version(
-            format!(
-                "{}\n\
-                 BLS library: {}\n\
-                 SHA256 hardware acceleration: {}\n\
-                 Specs: mainnet (true), minimal ({})",
-                 VERSION.replace("Lighthouse/", ""),
-                 bls_library_name(),
-                 have_sha_extensions(),
-                 cfg!(feature = "spec-minimal"),
-            ).as_str()
+            long_version
         )
         .arg(
-            Arg::with_name("spec")
-                .short("s")
-                .long("spec")
-                .value_name("DEPRECATED")
-                .help("This flag is deprecated, it will be disallowed in a future release. This \
-                    value is now derived from the --network or --testnet-dir flags.")
-                .takes_value(true)
+            Arg::new(CONFIG_FILE_FLAG)
+                .long(CONFIG_FILE_FLAG)
+                .help(
+                    "The filepath to a YAML or TOML file with flag values. The filename must \
+                     end in `.toml`, `.yml`, or `.yaml`. To override any options in \
+                    the config file, specify the same option in the command line."
+                )
                 .global(true)
-        )
+                .takes_value(true),
+        ).arg(
+        Arg::new(SPEC_FLAG)
+            .short('s')
+            .long(SPEC_FLAG)
+            .value_name("DEPRECATED")
+            .help("This flag is deprecated, it will be disallowed in a future release. This \
+                    value is now derived from the --network or --testnet-dir flags.")
+            .takes_value(true)
+            .global(true)
+    )
```

Comment:
Another difference fmt does not pick. It seems to be it nopes out of very large functions.

```suggestion
        )
        .arg(
            Arg::new(SPEC_FLAG)
                .short('s')
                .long(SPEC_FLAG)
                .value_name("DEPRECATED")
                .help("This flag is deprecated, it will be disallowed in a future release. This \
                        value is now derived from the --network or --testnet-dir flags.")
                .takes_value(true)
                .global(true)
        )
```